### DISCUSSION TITLE
Using javascript worker causes warnings with correct javascript code.

### DISCUSSION THREAD

**Discussion by velimir-97** [ajaxorg/ace#5119] [Q&A]
Hello,

I'm using the ace editor and when in javascript mode and `useWorker` is set to `true` I'm getting warnings like 
`'async functions' is only available in ES8 (use 'esversion: 8').`

To test it out you can paste `async function x() { }` on https://ace.c9.io/build/kitchen-sink.html

I haven't seen anywhere where to set `esversion` to `8`, does anyone know how to resolve this?

**Comment by JosephShown** [comment]
To use ES8 features in your JavaScript code, you need to make sure that your runtime environment supports ES8. Most modern browsers and Node.js versions support ES8, so you should be able to use ES8 features without any additional configuration.

If you’re using a transpiler like Babel to compile your JavaScript code, you may need to configure it to support ES8 features. You can do this by installing the appropriate presets and plugins and adding them to your Babel configuration.

For example, to use async/await in your code with Babel, you would need to install the @babel/plugin-transform-async-to-generator plugin and add it to your Babel configuration like this:

{
  "plugins": ["@babel/plugin-transform-async-to-generator"]
}

**Comment by DontShootMe** [comment]
This error come from Options passed to JsLint.
JsLint is embed in the javascript worker, and it have the option : esnext = true
It is why you have this warning.

But i don't know how to modify this options without a build step.
Maybe it will be a good thing if it become possible to pass this kind of options at the editor configuration step to the worker
